js/bybit.js

```diff
@@ -2751,43 +2885,26 @@ module.exports = class bybit extends Exchange {
             // 'stop_order_id': id, // only for conditional orders
             // 'p_r_trigger_price': 123.45, // new trigger price also known as stop_px
         };
-        let method = undefined;
-        if (market['swap']) {
-            if (market['linear']) {
-                method = 'privateLinearPostOrderReplace';
-            } else if (market['inverse']) {
-                method = 'v2PrivatePostOrderReplace';
-            }
-        } else if (market['future']) {
-            method = 'futuresPrivatePostOrderReplace';
-        }
-        const stopOrderId = this.safeString (params, 'stop_order_id');
-        if (stopOrderId !== undefined) {
-            if (market['swap']) {
-                if (market['linear']) {
-                    method = 'privateLinearPostStopOrderReplace';
-                } else if (market['inverse']) {
-                    method = 'v2PrivatePostStopOrderReplace';
-                }
-            } else if (market['future']) {
-                method = 'futuresPrivatePostStopOrderReplace';
-            }
-            request['stop_order_id'] = stopOrderId;
-            params = this.omit (params, [ 'stop_order_id' ]);
-        } else {
-            request['order_id'] = id;
-        }
+        const orderType = this.safeString (params, 'orderType');
+        const isStop = this.safeValue (params, 'stop', false);
+        const isConditionalOrder = isStop || (orderType === 'stop' || orderType === 'conditional');
+        params = this.omit (params, [ 'orderType', 'stop' ]);
+        const idKey = isConditionalOrder ? 'stop_order_id' : 'order_id';
+        request[idKey] = id;
         if (amount !== undefined) {
-            let qty = this.amountToPrecision (symbol, amount);
-            if (market['inverse']) {
-                qty = parseInt (qty);
-            } else {
-                qty = parseFloat (qty);
-            }
-            request['p_r_qty'] = qty;
+            request['p_r_qty'] = amount;
         }
         if (price !== undefined) {
-            request['p_r_price'] = parseFloat (this.priceToPrecision (symbol, price));
+            request['p_r_price'] = price;
```

Comment:
Same here, did we miss `exchange.priceToPrecision` ?

Reply:
Same as above